generalhw/adding-vnc-setup.md

```diff
@@ -189,80 +189,37 @@ systemctl enable --now kvmd-vnc
+output of `v4l2-ctl --list-formats`).
 
 In this case, copy `kvmd.streamer.cmd` part of `/etc/kvmd/main.yaml` to
-`/etc/kvmd/override.yaml` and change `--format=` parameter to `RGB24` and
-`--encoder=` to `cpu`.  This part will look like this:
+`/etc/kvmd/override.yaml` and change `--format=` parameter to `rgb24` and
+`--encoder=` to `cpu`.  Due to [bug in TC358743](https://github.com/raspberrypi/linux/issues/6068),
+red and blue channels will be swapped.  It isn't enough to use `bgr24` - it
+would be properly communicated to the controller, so it will swap the channels
+as well as metadata about used format.  To handle this problem, a new option
+`--format-swap-rgb` has to be added.
+
+Part of `override.yaml` responsible for setting proper video format will look
+like this:
 
 ```yaml
 kvmd:
     streamer:
         cmd:
             - "/usr/bin/ustreamer"
             # ...
-            - "--format=RGB24"
+            - "--format=rgb24"
             - "--encoder=cpu"
+            - "--format-swap-rgb"
             # ...
 ```
 
-Apply the changes by restarting `kvmd` with `systemctl restart kvmd`.
-
-Now, if colors are completely off now (e.g., red is green), that might be due
-to `RGB` being treated as `BGR` or vice versa.  Need to patch `ustreamer`...
-
-1. Clone [ustreamer].
-2. Checkout tag that correspond to appropriate version, so it should be
-   compatible with the rest of PiKVM.
-3. Apply the following patch:
-
-```patch
-diff --git a/src/ustreamer/encoders/cpu/encoder.c b/src/ustreamer/encoders/cpu/encoder.c
-index 1c20e6b..a56a7f2 100644
---- a/src/ustreamer/encoders/cpu/encoder.c
-+++ b/src/ustreamer/encoders/cpu/encoder.c
-@@ -221,15 +221,29 @@ static void _jpeg_write_scanlines_rgb565(struct jpeg_compress_struct *jpeg, cons
- }
- 
- static void _jpeg_write_scanlines_rgb24(struct jpeg_compress_struct *jpeg, const us_frame_s *frame) {
-+	uint8_t *line_buf;
-+	US_CALLOC(line_buf, frame->width * 3);
-+
- 	const unsigned padding = us_frame_get_padding(frame);
- 	uint8_t *data = frame->data;
- 
- 	while (jpeg->next_scanline < frame->height) {
--		JSAMPROW scanlines[1] = {data};
--		jpeg_write_scanlines(jpeg, scanlines, 1);
-+		uint8_t *ptr = line_buf;
-+
-+		for (unsigned x = 0; x < frame->width; ++x) {
-+			*(ptr++) = data[2];
-+			*(ptr++) = data[1];
-+			*(ptr++) = data[0];
- 
--		data += (frame->width * 3) + padding;
-+			data += 3;
-+		}
-+		data += padding;
-+
-+		JSAMPROW scanlines[1] = {line_buf};
-+		jpeg_write_scanlines(jpeg, scanlines, 1);
- 	}
-+
-+	free(line_buf);
- }
- 
- #define JPEG_OUTPUT_BUFFER_SIZE ((size_t)4096)
-```
-
-4. Compile: `make apps`
-5. Backup original: `cp /usr/bin/ustreamer /usr/bin/ustreamer.bak`
-6. Install modified version: `cp -f src/ustreamer.bin /usr/bin/ustreamer`
-7. Restart the service which executes `ustreamer`: `systemctl restart kvmd`
+The snippet above only shows modified streamer command options, but all existing
+ones have to be copied as well.  To obtain them, use `kvmd -m`.
```

Comment:
```suggestion
ones have to be copied as well.
```
The values are copied from `/etc/kvmd/override.yaml`, no need for `kvmd -m`.

Reply:
I think you meant `/etc/kvmd/main.yaml`, but anyway, I skipped over that sentence, thanks for catching it.

https://github.com/3mdeb/openqa-tests-qubesos/compare/c91f3f4dccd0ec59c0a56ad523bb3f9430a1a6df..fee8141d25c417c5b87babaa21c665c0f035762f